index.js

```diff
@@ -32,13 +32,14 @@ function handleArgs(cmd, args, opts) {
 		maxBuffer: TEN_MEGABYTES,
 		stripEof: true,
 		preferLocal: true,
+		localDir: parsed.options.cwd || process.cwd(),
 		encoding: 'utf8',
 		reject: true,
 		cleanup: true
 	}, parsed.options);
 
 	if (opts.preferLocal) {
-		opts.env = npmRunPath.env(opts);
+		opts.env = npmRunPath.env(Object.assign({}, opts, {cwd: opts.localDir}));
```

Comment:
@sindresorhus I defined the default for `localDir` up above, but we still need the `Object.assign` call here, because `npmRunPath` needs `cwd` to be set to define where it looks, and if we override `cwd` with the value of `opts.localDir` it would mess up the execution of the actual command.